### PR TITLE
Migrating to KeyboardEvent.code

### PR DESCRIPTION
**Pull request by Mephistic**
This updates all uses of `event.keyCode` to use the newer, un-deprecated `KeyboardEvent` and should resolve https://github.com/pokeclicker/pokeclicker/issues/747.

Testing:
* I tested these changes in the most recent Chrome/Firefox/Edge and all seems fine.

One nit: It seems like we need to use `e.originalEvent.code` instead of just `e.code` because of the way TypeScript plays with https://github.com/jquery/jquery/issues/3978 - the `code` property will be available on the event object when debugging, but it doesn't seem to register as available when compiling.

### CODE REVIEW

File: src/scripts/GameController.ts [hunk 1]
```diff
@@ -115,33 +108,33 @@ class GameController {
     }
 
     static addKeyListeners() {
-        $(document).on('keydown', function (e) {
+        $(document).keydown(function (e) {
```
Comment (Aegyo) [inline src/scripts/GameController.ts:111]: Event aliases are deprecated, we should be using `.on(eventName`

**Comment by Aegyo** [comment]
Seems like the type definitions we are using for jquery are missing the the `code` property for events. It was added to the jquery event object over 2 years ago.

I wonder how long it would take to get a pr approved over at DefinitelyTyped...

**Comment by Mephistic** [comment]
Let's find out: https://github.com/DefinitelyTyped/DefinitelyTyped/pull/48201

**Comment by Mephistic** [comment]
Looks like the DefinitelyTyped PR was merged so we can just use `e.code`

**Review by Aegyo — APPROVED** [review]

**Pull request merged**